active-rfcs/0000-relative-url-scheme.md

```diff
@@ -0,0 +1,95 @@
+- Start Date: 2021-12-20
+- Reference Issues: [#46](https://github.com/withastro/rfcs/pull/46)
+- Implementation PR: <!-- leave this empty -->
+
+# Summary
+
+Astro should support source-relative URL resolution in the HTML block of Astro files.
+
+# Example
+
+To resolve the source-relative URL of an image, use a `local:` prefixed URL.
+
+```html
+<img src="local:kitten.avif" alt="Kitten" />
+```
+
+# Motivation
+
+In `.astro` files, it’s not always easy to know what a relative URL is relative to.
+
+```astro
+<img src="kitten.avif" alt="kitty" />
+```
+
+In this example, is `kitten.avif` relative to the output `.html` or the source `.astro`? Or, if this example is found within a component `.astro` file, is it relative to that component file?
+
+This ambiguity raises a need for:
+
+- An intuitive way to author URLs relative to a source file.
+- An intuitive way to author URLs within the HTML of an `.astro` file.
+
+# Detailed design
+
+A `local` scheme instructs Astro to resolve the URL relative to the source file.
+
+```astro
+<img srcset="local:kitten.avif 1x, local:kitten@2x.avif" alt="Kitten" />
+```
+
+```astro
+<picture>
+  <source srcset="local:media/kitten@2x.avif" media="(min-width: 800px)" />
+  <img src="local:media/kitten.jpg" alt="" />
+</picture>
+```
+
+```astro
+<div style="background-image: url(local:kitten-background.avif);"></div>
+```
+
+```astro
+<div data-my-framework-attr-for-image="local:kitten-background.avif"></div>
+```
+
+# Drawbacks
+
+All attribute values need to be checked for `local:`. See alternatives below
+
+The provided examples are intuitive and convenient for authors, but not immediately obvious to an implementer.
+
+# Alternatives
+
+- Limit support to `local:` prefixed attributes
+  ```html
+  <img local:src="kitten.avif" alt="Kitten" />
+  ```
+  - It’s not obvious if this would be more performant than checking every attribute for a `local:` prefix.
```

Comment:
I think it probably is, you only have to check the first 6 characters of each attribute rather than checking for 6 characters anywhere in an N character string.

Reply:
I'm not worried about the performance of doing it as a URL scheme, to be clear, just nothing that I think it isn't as performant as an attribute would be.

Reply:
I agree. Plus, if it were `local:src` then we still have to check the value, unless we drop support for several of the common use cases (`srcset`, `data-`).

Reply:
This note has been removed from the RFC.